### ISSUE TITLE
Editing the attributes of an owner inclusion

### ISSUE THREAD

**Issue by bmarchal54** [sympa-community/sympa#989]
When you want to modify the attributes associated with an inclusion of owners by means of a list, the modifications are not taken into account.

Version
-------
6.2.56

Installation method
-------------------
Installed with rpm

Expected behavior
-----------------
In the web interface, we want, for example, to change a list of owners from 'normal' to 'privileged', or from 'conceal' to 'noconceal', or vice versa ...
The modifications are not taken into account at the level of the list of owners ('users' menu), nor in the database, although the configuration file is indeed modified
My current solution is to remove the inclusion and then recreate it.

Actual behavior
---------------

Additional information
----------------------

**Comment by ikedas** [comment]
I think it's a bug.  Attributes of included owners should not be modified with web UI.

**Comment by ikedas** [comment]
@bmarchal54 , I couldn't reproduce the behavior you reported: I could change the attributes of owners who have been included using web UI.
Could you please tell us how did you operate to reproduce?

**Comment by bmarchal54** [comment]
Hello,
I am on leave for two weeks without access to a computer (except for mail).
I will then send you the various operations that I perform.
I will keep you posted as soon as possible.
cordially

---
Benoît Marchal
Sous-direction des infrastructures
Direction du Numérique
Université de Lorraine
Benoit.Marchal@univ-lorraine.fr

**Comment by bmarchal54** [comment]
Hello 

So I have a list whose owners are defined by an external data source (a list inclusion). 
When creating this data source, I leave the default values (noconceal, nomail). 
I want to change noconceal to conceal, so I go back to setting the data source to change this value. I modify, I validate. 
I have the change in the 'config' file, but the database is not modified (table 'admin_table' , field 'visibility_admin' which remains at noconceal). 

cordially 

--- 
Benoît MARCHAL 
Sous direction Infrastructure & Services 
Direction du Numérique 
Université de Lorraine 
Tél : 03.72.74.11.39 / 06.07.35.50.65 
mél : Benoit.Marchal@univ-lorraine.fr

**Comment by ikedas** [comment]
I understood what you feel unintuitive.

OTOH I feel the user options (`reception` etc.) by each data source are a bit complicating.  How about changing the behavior making it the same as subscribers? Such as:
  - Remove user options (`reception`, `visibility` and `privilege`) by each data source parameters;
  - Add a new list parameter `default_owner_options` including user options above, analogous to [`default_user_options`](https://sympa-community.github.io/gpldoc/man/sympa_config.5.html#default_user_options) (Subscription profile);
  - And, like subscribers, user options specified by the parameter above will be applied only at the time of the first inclusion of the owner.

**Comment by bmarchal54** [comment]
Hello, 
In our use, for the moment, all the owners of the same inclusion have the same profile. On the other hand, this is not the case for two different inclusions. 
A little more details: in our university, we have local IT services for the components. Each component has its own service. When the component requests a list, we are able automatically to determine the associated IT department and we systematically set it to 'privileged', 'conceal' and 'nomail'. Some components require additional inclusions which have the default options 'normal', 'noconceal' and 'mail'. 
With default options, this will be valid regardless of the inclusions. When a new subscriber appears in one of the inclusions, what options will he have? Those by default which will therefore not necessarily be those required for this inclusion. We should therefore reconsider our options, but we will not know. 
The current function pleased me well. It is during the definition of the inclusion that we determine the default options for all the members of this inclusion, options which should be able to be modified over time. 
Have a good day 

--- 
Benoît MARCHAL 
Sous direction Infrastructure & Services 
Direction du Numérique 
Université de Lorraine 
Tél : 03.72.74.11.39 / 06.07.35.50.65 
mél : Benoit.Marchal@univ-lorraine.fr

**Comment by bmarchal54** [comment]
Hello, 
Since the patch passed to take into account the privileged owners (or since the passage in 6.2.56, I do not know exactly), there is a side effect: 
the person's attributes in the database are no longer correctly populated in the table 'admin-table 'when the inclusion comes from an external data source. ie. the 'include_admin' and 'include_sources_admin' fields are NULL while the first should be 1 and the second should be a value other than NULL. 
I don't know if this can help you find the problem. 
cordially 

--- 
Benoît MARCHAL 
Sous direction Infrastructure & Services 
Direction du Numérique 
Université de Lorraine 
Tél : 03.72.74.11.39 / 06.07.35.50.65 
mél : Benoit.Marchal@univ-lorraine.fr

**Comment by ikedas** [comment]
@bmarchal54 ,
`include_admin` and `include_sources_admin` fields were deprecated by Sympa 6.2.46 (ae01bc3).  Recently new fields including `inclusion_admin` and `inclusion_label_admin` are used.

**Comment by mathieuroquefort** [comment]
Hello,
We have the same problem here but it occurs differently. Here is how :

We use families for all of our lists.
In the instanciation file of each list lies this :
```
	<owner_include multiple="1">
		<source>listmaster</source>
		<profile>privileged</profile>
		<reception>nomail</reception>
		<visibility>conceal</visibility>
	</owner_include>
	<editor_include multiple="1">
		<source>listmaster</source>
		<reception>nomail</reception>
		<visibility>conceal</visibility>
	</editor_include>
```
After creation, the address into the file `listmaster` is imported, but its settings (privileged, nomail, concealed) are not. We have to set all of them manually.
So, the problems comes not only after modification, but also upon creation. We use sympa v6.2.58, and I've had this problem since v6.1 (which was our first 6.x use).


> OTOH I feel the user options (`reception` etc.) by each data source are a bit complicating. How about changing the behavior making it the same as subscribers? Such as:
> [...]
>     * And, like subscribers, user options specified by the parameter above will be applied only at the time of the first inclusion of the owner.

This behaviour is exactly the current one for us, and that is the problem.
So, it wouldn't help at all.
These user options are very useful. Please keep them.

**Comment by ikedas** [comment]
Hi @mathieuroquefort ,

Unlike `<owner>` or `<editor>`, `<owner_include>` and `<editor_include>` elements may not contain attributes `<profile>`, `<reception>` and/or `<visibility>`.

**Comment by ikedas** [comment]
@mathieuroquefort ,
You may have to edit `config.tt2` file in the list creation template you are using, for example:
``` code
--- a/create_list_templates/discussion_list/config.tt2
+++ b/create_list_templates/discussion_list/config.tt2
@@ -26,6 +26,9 @@ owner
 [% FOREACH o = owner_include -%]
 owner_include
   source [% o.source %]
+  [% IF o.profile %]profile [% o.profile %][% END -%]
+  [% IF o.reception %]reception [% o.reception %][% END -%]
+  [% IF o.visibility %]visibility [% o.visibility %][% END -%]

 [% END -%]
 [% END -%]
```

**Comment by ikedas** [comment]
PR#1251 added some improvements:
  - Now the owners & the moderators also have `default_owner_options` and `default_editor_options` parameters, the counterparts of `default_user_options` for the subscribers.
  - Now `member_include` parameter also has `reception` & `visibility` subparameters, the counterparts of them in `owner_include` and `editorinclude`.

Then,
  - When the owners and/or editors are added (manually or by inclusion), attributes in `default_*_options` parameters will be used.
  - However, when they are added by inclusion, attributes in `*_include`  parameters may override the attributes above.
 
This feature will be included in the next beta, 6.2.67b.2.  We are appreciated if you will check it.  Thanks!

**Comment by bmarchal54** [comment]
Hello, 

I do not understand correctly: all owners / moderators will have the same options for the same list? IE. if I choose the default 'privileged' option for a list, all owners will be privileged owners, right? And in this case, how am I going to differentiate the rights I want to give to different people on a list. 

We have many lists (more than 10,000) and we automatically add, by list inclusion (owner_include) lists for which, the subscribers of these are privileged owners and for others only simple owners. Of course, they are not all the same for the whole: we have a division into component and according to this one, we add a list, rather than another. 

Will this always be possible with your proposal, without going over each one? 

Good day 

--- 
Benoît MARCHAL 
Sous direction Infrastructure & Services 
Direction du Numérique 
Université de Lorraine 
Tél : 03.72.74.11.39 / 06.07.35.50.65 
mél : ***@***.***

**Comment by ikedas** [comment]
> Hello, I do not understand correctly: all owners / moderators will have the same options for the same list? IE. if I choose the default 'privileged' option for a list, all owners will be privileged owners, right? And in this case, how am I going to differentiate the rights I want to give to different people on a list.

Yes, all owners / moderators will have the same options for the same list, _if_ the options were not specified explicitly, e.g. the owners were included from data source.

In such case, now the options can be defined in advance, instead of using fixed options.

> We have many lists (more than 10,000) and we automatically add, by list inclusion (owner_include) lists for which, the subscribers of these are privileged owners and for others only simple owners. Of course, they are not all the same for the whole: we have a division into component and according to this one, we add a list, rather than another.
> 
> Will this always be possible with your proposal, without going over each one? 

I think that what you are able to do now should be able to do.

If you have any doubts, could you please show us what specific settings you have.
